File: Model/PlaylistModel.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace KPCourseWork.Models;

[Table("playlist")]
public class PlaylistModel
{
    [Column("id")]
    public int Id { get; set; }
    [Column("name")]
    public string Name { get; set; }
    [Column("imagePath")]
    public string? Image { get; set; }
    public List<TrackModel>? Tracks { get; set; }
    [Column("user_id")]
    public UserModel? User { get; set; }
    public DateTime createdAt { get; set; }
    public DateTime updatedAt { get; set; }
}

File: Frontend/src/app/models/playlist.model.ts
import {TrackModel} from "./track.model";
import {UserModel} from "./user.model";

export type PlaylistModel = {
  id: number
  name: string
  image: string
  tracks: TrackModel[]
  user: UserModel
  createdAt: string
  updatedAt: string
}

export type UpdatePlaylistModel = {
  name?: string
  image?: string
  tracks?: TrackModel[]
  user?: UserModel
  createdAt?: string
  updatedAt?: string
}
